### PR TITLE
Fixing OSS-Fuzz issues

### PR DESCRIPTION
**Pull request by xhanulik**
##### Checklist
- [ ] PKCS#11 module is tested
- [ ] Windows minidriver is tested
- [ ] macOS tokend is tested

### CODE REVIEW

File: src/pkcs15init/pkcs15-oberthur.c [hunk 1]
```diff
@@ -637,24 +645,39 @@ cosm_generate_key(struct sc_profile *profile, struct sc_pkcs15_card *p15card,
...
 	LOG_TEST_RET(ctx, rv, "cosm_generate_key() CARDCTL_OBERTHUR_GENERATE_KEY failed");
 
 	/* extract public key */
 	pubkey->algorithm = SC_ALGORITHM_RSA;
 	pubkey->u.rsa.modulus.len   = key_info->modulus_length / 8;
 	pubkey->u.rsa.modulus.data  = malloc(key_info->modulus_length / 8);
-	if (!pubkey->u.rsa.modulus.data)
+	if (!pubkey->u.rsa.modulus.data) {
+		sc_file_free(prkf);
+		sc_file_free(tmpf);
```
Comment (Jakuje) [inline src/pkcs15init/pkcs15-oberthur.c:668]: missing `free(args.pubkey);` I think. Same in the following block.

**Review by Jakuje — APPROVED** [review]

File: src/libopensc/muscle.c [hunk 2]
```diff
@@ -627,8 +627,10 @@ int msc_extract_rsa_public_key(sc_card_t *card,
 	if(!*modulus) LOG_FUNC_RETURN(card->ctx, SC_ERROR_OUT_OF_MEMORY);
 	memcpy(*modulus, buffer, *modLength);
 	*expLength = (buffer[*modLength] << 8) | buffer[*modLength + 1];
-	if (*expLength > sizeof buffer)
+	if (*expLength > sizeof buffer) {
+		free(*modulus); *modulus = NULL;
```
Comment (frankmorgner) [inline src/libopensc/muscle.c:631]: a line break would be nice, here.

**Review by frankmorgner — APPROVED** [review]
Does the security advisory need to be extended with this? AFAIKT, this only touches possible memory leaks.

**Comment by frankmorgner** [comment]
@xhanulik , could you check Packit's RPM build? It seems to be failing consistently due to some dependency problem.

**Comment by Jakuje** [comment]
> Does the security advisory need to be extended with this? AFAIKT, this only touches possible memory leaks.

Correct. The only recent security-related reports from oss-fuzz are already in verified state.

**Comment by Jakuje** [comment]
> @xhanulik , could you check Packit's RPM build? It seems to be failing consistently due to some dependency problem.

Verca is away so I will have a look into this issue in separate PR.

**Pull request merged**